src/ios/FileUploader.m

```diff
@@ -20,30 +21,48 @@ -(id)init{
         return nil;
     [UploadEvent setupStorage];
     self.responsesData = [[NSMutableDictionary alloc] init];
+    self.uploadStartTimes = [[NSMutableDictionary alloc] init];
     NSURLSessionConfiguration* configuration = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:[[NSBundle mainBundle] bundleIdentifier]];
     configuration.HTTPMaximumConnectionsPerHost = FileUploader.parallelUploadsLimit;
     configuration.sessionSendsLaunchEvents = NO;
     self.manager = [[AFURLSessionManager alloc] initWithSessionConfiguration:configuration];
     __weak FileUploader *weakSelf = self;
     [self.manager setTaskDidCompleteBlock:^(NSURLSession * _Nonnull session, NSURLSessionTask * _Nonnull task, NSError * _Nullable error) {
         NSString* uploadId = [NSURLProtocol propertyForKey:kUploadUUIDStrPropertyKey inRequest:task.originalRequest];
+        NSDate *startTime = weakSelf.uploadStartTimes[uploadId];
+        NSDate *endUploadTime = [NSDate date];
+        NSTimeInterval timeInterval = [endUploadTime timeIntervalSince1970];
+        long long uploadTimeinMS = (long long)(timeInterval * 1000);
+        NSTimeInterval duration = [endUploadTime timeIntervalSinceDate:startTime];
         NSLog(@"[BackgroundUpload] Task %@ completed with error %@", uploadId, error);
         if (!error){
             NSData* serverData = weakSelf.responsesData[@(task.taskIdentifier)];
             NSString* serverResponse = serverData ? [[NSString alloc] initWithData:serverData encoding:NSUTF8StringEncoding] : @"";
             [weakSelf.responsesData removeObjectForKey:@(task.taskIdentifier)];
-            [weakSelf saveAndSendEvent:@{
-                @"id" : uploadId,
-                @"state" : @"UPLOADED",
-                @"statusCode" : @(((NSHTTPURLResponse *)task.response).statusCode),
-                @"serverResponse" : serverResponse
-            }];
+            if (!isnan(duration)) {
```

Comment:
Discussed with @parveshneedhoo. We can simplify this by creating the dictionary with the required values first then add `uploadDuration`, `finishUploadTime` if the if passes